extensions/provisioning/verify.go

```diff
@@ -428,27 +429,28 @@ func VerifySnapshots(client *rancher.Client, localclusterID string, clusterName
 			return false, fmt.Errorf("no snapshots found")
 		}
 
-		// Indexed from 0 for S3 checks to ensure that the local backup location does not have the s3Prefix.
-		// Needed to ensure that the correct S3 snapshot is restored.
-		if strings.Contains(snapshotNameList[0], s3Prefix) {
-			snapshotToBeRestored = snapshotNameList[len(snapshotNameList)-1]
-			return true, nil
-		}
+		if strings.Contains(fmt.Sprintf("%v", snapshotNameList), s3Prefix) {
```

Comment:
what if the cluster has s3 enabled, but we're trying to test local etcd snapshots/restores?

Reply:
I don't think that is a valid use-case. If you have S3 enabled on a cluster and you perform an etcd snapshot, the expectation is you are going to use cloud storage for backup/restore.

Reply:
Resolved.